changelog/unreleased/bugfixes/7030.md

```diff
@@ -0,0 +1 @@
+- Fixed "global reference table overflow" in case an application accumulates and keeps links to `RouteProgress#upcomingRoadObjects`.
```

Comment:
```suggestion
- Fixed "global reference table overflow" by reducing count of transfered objects through the JNI
```

Reply:
That's Łukasz's original phrasing. And I think it's accurate. If the app doesn't accumulate the links, nothing really changes.

Reply:
Original PR from Łukasz coved only then case when app accumulates links. Current fix also addresses the case when GC can't remove transferred alters fast enough, so it's more general, while original was more specific. Does it make sense?